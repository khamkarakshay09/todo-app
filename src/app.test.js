import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import store from "./redux/store";
import App from "./app";

describe("App component", () => {
  const wrapper = mount(
    <Provider store={store}>
      <App />
    </Provider>
  );
  const inputElement = wrapper.find("input#qa-todo-input");
  const submitBtn = wrapper.find("button#qa-submit-btn");
  it("Snapshot Test for App component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Todo Form", () => {
    expect(inputElement).toHaveLength(1);
    expect(submitBtn).toHaveLength(1);
  });
});
