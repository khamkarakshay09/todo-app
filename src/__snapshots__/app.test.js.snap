// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Snapshot Test for App component 1`] = `
<div
  className="MuiContainer-root makeStyles-root-1 MuiContainer-maxWidthLg"
>
  <section>
    <h3
      className="MuiTypography-root makeStyles-header-2 MuiTypography-h3 MuiTypography-gutterBottom"
    >
      Todos
    </h3>
    <form
      className="makeStyles-root-3"
      onSubmit={[Function]}
    >
      <div
        className="MuiFormControl-root MuiTextField-root makeStyles-todoInput-4"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="qa-todo-input"
          id="qa-todo-input-label"
        >
          Add New Task
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="qa-todo-input"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-7"
            >
              <span>
                Add New Task
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
        disabled={false}
        id="qa-submit-btn"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </form>
    <div
      className="MuiPaper-root makeStyles-root-9 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiPaper-root MuiCard-root makeStyles-noTaskCard-11 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          className="MuiCardContent-root"
        >
          No tasks found!
        </div>
      </div>
      <div
        className="MuiPaper-root MuiCard-root makeStyles-filters-12 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <p
          className="makeStyles-count-13"
        >
          All(
          0
          )
        </p>
        <div
          className="makeStyles-btnGrp-14"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-active-15"
            disabled={false}
            id="All"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              All
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text"
            disabled={false}
            id="Active"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Active
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text"
            disabled={false}
            id="Completed"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Completed
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </section>
</div>
`;
